Validators, and Query scanning

ladal1 opened the pull request.

ladal1 committed:
Deleted unnecessary imports of ensure_tuple

ladal1 committed:
Rename of Validator in test to reflect what it does

ladal1 committed:
Django validator and Foreign key validator implemented

ladal1 committed:
Additional workflows for python 3. versions

ladal1 committed:
Print in converter.py missed

karlosss commented on .github/workflows/run_tests(36).yml:
```diff
@@ -0,0 +1,30 @@
+name: Run tests (3.6)
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.6]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Run Tests
+      run: |
+        ./run_tests.sh
```
dafuq? opravdu se to musi delat takhle blbe, ze xkrat zkopirujes stejny soubor? to si nemyslim...

ladal1 replied:
Asi ne, to jsem udělal rychle abych nezapoměl že to chci udělat, měl jsem v plánu to ještě předělat než ti to pošlu, ta změna z draftu byla předtím než jsem tohle přidal

ladal1 replied:
Spojené do pole, a nastavení aby se tak workflow jmenoval s proměnnými

karlosss replied:
tohle vypada mnohem lepsi, jestli to funguje

ladal1 replied:
Mělo by, viz akce u tohohle posledního pushe

karlosss commented on simple_api/django_object/converter.py:
```diff
@@ -54,25 +56,29 @@ def convert_django_field(field, field_name, both_fields, input_fields, output_fi
+    validators.append(ForeignKeyValidator(target_model))
+    input_fields[field_name + "_id"] = converted_pk_field(nullable=field.null, exclude_filters=(),
+                                                          validators=validators)
+    output_fields[field_name] = ObjectType(target_model, nullable=field.null, exclude_filters=(), validators=validators)
```
output field potrebuje validatory? asi to nevadi, ale je to zbytecne a pri cteni kodu muze byt matouci

ladal1 replied:
Nepotřebuje, byla to jedna věc co jsem přehlédl že jsem pokazil když jsem se zbavoval commitu tak se změnila...

karlosss commented on simple_api/django_object/validators.py:
```diff
@@ -0,0 +1,29 @@
+        self.target_model = target_model
+
+    def validation_statement(self, request, value=None, **kwargs):
+        return self.target_model.objects.filter(id=value).exists()
```
preferuju pouzivat `pk` misto `id` - `id` je builtin funkce v pythonu

karlosss commented on simple_api/django_object/validators.py:
```diff
@@ -0,0 +1,29 @@
+
+
+class DjangoValidator(Validator):
+    def __init__(self, field_name, validation_fn):
```
`field_name` sem nepatri - pokud validator potrebuje vedet, jake validuje policko, tak to do nej injektuj nejak jinak nez pres konstruktor, ale stejne si myslim, ze to vedet nepotrebuje

karlosss commented on simple_api/django_object/validators.py:
```diff
@@ -0,0 +1,29 @@
+            return False
+
+    def error_message(self, **kwargs):
+        return "Django validator failed in field {}".format(self.field_name)
```
Toto by graphql melo umet samo od sebe priradit ke konkretnimu fieldu

ladal1 replied:
Nevím jak by to vymyslela, protože tak teď nezjistí z toho které políčko to v té funkci z build_validation_fn pokazilo

karlosss commented on simple_api/django_object/validators.py:
```diff
@@ -0,0 +1,29 @@
+
+class DjangoValidator(Validator):
+    def __init__(self, field_name, validation_fn):
+        self.fn = validation_fn
```
`self.callable = callable` - nemusi to byt nutne funkce

ladal1 replied:
Přejmenováno

karlosss commented on simple_api/django_object/validators.py:
```diff
@@ -0,0 +1,29 @@
+        return "Django validator failed in field {}".format(self.field_name)
+
+
+class ForeignKeyValidator(Validator):
```
zatim asi OK takhle, podle me se casem ukaze, ze je to prilis coarse-grained

ladal1 replied:
Possible, uvidíme

karlosss commented on simple_api/object/validators.py:
```diff
@@ -39,12 +38,15 @@ def fn(**kwargs):
             for validator in validators:
-                if not validator.validation_statement(kwargs["request"], value=kwargs["params"][param_field]):
+                if param_field in kwargs["params"] and \
+                        not validator.validation_statement(kwargs["request"], value=kwargs["params"][param_field]):
```
potom prosim sjednot API s permissions:

```python
def is_valid(...):
    return self.validation_statement(...)
```

a potom `validator.is_valid(...)`

ladal1 committed:
Combined runtests into one workflow file

ladal1 committed:
Fixes based upon PR discussion

ladal1 committed:
LogicalConnectors first draft

karlosss committed:
change metadata format example, TODO

ladal1 committed:
Frontend base

ladal1 committed:
urlpatterns and settings for frontend split into different branch

karlosss committed:
change metadata format

ladal1 committed:
Actual badge in README.md

ladal1 committed:
Stop fail-fast on different python versions

ladal1 committed:
Merge pull request #1 from karlosss/change-metadata

Change metadata merge, now to fixing tests

ladal1 committed:
Separate frontend to different repo

ladal1 committed:
Remove build_patterns from test_utils

ladal1 committed:
Merge remote-tracking branch 'origin/master'

ladal1 committed:
FIX: Actions not contained in object could not have data

ladal1 committed:
Security query analysis and introspection middleware block

ladal1 committed:
Add mutations and defaults into metaschema

ladal1 committed:
Disable introspection middleware

ladal1 committed:
Weight, depth and list check before resolvers

ladal1 committed:
Validators include error field if they know it

ladal1 committed:
Missed a single check for if the limit is set before raising exception on no limit on paginated data

ladal1 committed:
Moved the list_limit existence check

ladal1 committed:
Change build_patterns imports to new location

ladal1 committed:
Update metaschema test to new query

ladal1 committed:
Fixed tests for new build_patterns and new metaschema contents

ladal1 committed:
Forgot to remove simple_api_admin

ladal1 committed:
Fix of error caused by wrong indent and upped the testing settings score limit

ladal1 committed:
Change of score calculation, selections are added up to action weight not multiplied by it

ladal1 committed:
Tests for load protection query checks

ladal1 committed:
Ability to pass middleware to build_patters_w and allow general GraphQL introspection

ladal1 committed:
README update to new location of build_patterns

ladal1 committed:
Default testproject update so it doesn't become a relic of old design

ladal1 committed:
Add Actions limit, to protect against batching attacks instead of DoS

ladal1 committed:
Introspection wasn't being disabled as it should have been

ladal1 committed:
Passing of field weight scores from DjangoObject and only operation called is checked for limits

ladal1 committed:
More fields in filtering introspection, QL instead of Ql

ladal1 committed:
Settings update for test project

ladal1 committed:
Shelf field special weight removed

ladal1 committed:
Customizable default field difficulty

ladal1 committed:
Action limit removed because it wasn't useful at all (simple weight schema is clearer)

ladal1 committed:
action limit was hiding in another spot

ladal1 committed:
Added documentation source and result


Documentation push


Submodule connection


remove documentation


Documentation submodule

remove documentation


gitmodules


Documentation commit


Try to track the master branch


Gitmodules

ladal1 committed:
Documentation update


Documentation update
